src/Data/Tree/AVL/Internal.hs

```diff
@@ -82,20 +82,23 @@ data MapLayer h k v self
   | MLEmpty
     { _mlHash     :: h
     }
-    deriving (Eq, Functor, Foldable, Traversable, Generic, Binary)
+    deriving (Eq, Functor, Foldable, Traversable, Generic)
 
 type Map h k v = Free (MapLayer h k v) h
 
-deriving instance                                      Generic (Free t a)
-deriving instance (Binary (t (Free t a)), Binary a) => Binary  (Free t a)
+deriving instance Generic (Free t a)
 
 makeLenses ''MapLayer
 
+class Serialisable a where
+    serialise   :: a -> ByteString
+    deserialise :: ByteString -> Maybe a
```

Comment:
Please use `Either t` for some readable `t`. It will definitely help in the future to debug problems related to different app versions (or testnet/mainnet/cluster magic mismatch). A short class description (as a good practice, I know it's obvious what this class does) would be also great.